### ISSUE TITLE
Sentry plugin causes a crash when the logger is busy

### ISSUE THREAD

**Issue by etelyatn** [getsentry/sentry-unreal#418]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which version of the SDK?
0.11.0-0.12.0

How did you install the package? (Git-URL, Assetstore)
Git

Which version of Unreal?
5.2.1 source

Is this happening in Unreal (editor) or on a player like Android, iOS, Windows?
Editor, Windows, Windows Server

### Steps to Reproduce

When some process writes to the logs and at the same time Sentry tries to capture an event or message, it crashes.

```
write_json_str
Fatal Error: EXCEPTION_ACCESS_VIOLATION_READ / 0xf59000000
```

### Actual Result

![image](https://github.com/getsentry/sentry-unreal/assets/9371310/c5df0ac5-b28f-4b05-ba8a-37563d7306a9)

![image](https://github.com/getsentry/sentry-unreal/assets/9371310/6a797b0a-df64-40eb-b63b-fe3e17060749)

**Comment by tustanivsky** [comment]
@etelyatn Can you elaborate on how this issue can be reproduced?

**Comment by etelyatn** [comment]
@tustanivsky So let's say in our project the error is easily reproduced, a large amount of functionality + an active Log Viewer Pro plugin, at the time of sending a message to Sentry we receive this error. 

But I can’t reproduce it on an empty project, perhaps you can try to lock the file manually.

In general the plugin should not crash the app, even if it can not write a log, it must not crash.
So maybe some additional checks can help.

Thanks

**Comment by tustanivsky** [comment]
@etelyatn Can you share some examples of how you create and configure the new event before capturing it? Also, do you set any additional context data via the `USentrySubsystem`/`USentryScope` APIs? I think that the event's data was invalid or corrupted at some point which led to a crash upon its serialization rather than the issue being related to some background logging.

**Comment by etelyatn** [comment]
@tustanivsky Here is our usage

```c
void MyLog::SetTrackingUser(const FString& Id, const FString& Username, const FString& Email)
{
#ifdef ENABLE_SENTRY
	if (USentrySubsystem* Sentry = GetSentrySubsystem())
	{
		if (Sentry->IsEnabled())
		{
			if (USentryUser* SentryUser = NewObject<USentryUser>())
			{
				SentryUser->SetId(Id);
				SentryUser->SetEmail(Email);
				SentryUser->SetUsername(Username);
				Sentry->SetUser(SentryUser);
				
				FMsg::Logf(nullptr, 0, TEXT("MyLog"), ELogVerbosity::Display, TEXT("Set Sentry User: %s"), *Username);
			}
		}
	}
#endif
}
```

```c
bool MyLog::SentryLog(const FName& CategoryName, const FString& Msg, const ELogVerbosity::Type Verbosity)
{
	const ESentryLevel Level = ConvertVerbosityToSentryLevel(Verbosity);

	if (USentrySubsystem* Sentry = GetSentrySubsystem())
	{
		if (Sentry->IsEnabled())
		{
			Sentry->SetTag("category", CategoryName.ToString());
			Sentry->SetLevel(Level);
			// Sentry->CaptureMessage(Msg, Level);

			USentryEvent* Event = NewObject<USentryEvent>();
			Event->SetMessage(Msg);
			// Event->SetLevel(Level);

			return Sentry->CaptureEvent(Event) ? true : false;
		}
	}

	return false;
}
```

**Comment by tustanivsky** [comment]
@etelyatn I've just opened the PR with a fix for this issue (#445) which should be available in the next plugin release.
